assets/js/modules/analytics/datastore/service.js

```diff
@@ -50,22 +50,28 @@ export const selectors = {
 	 */
 	getServiceURL: createRegistrySelector(
 		( select ) => ( state, { path, query } = {} ) => {
-			const userEmail = select( CORE_USER ).getEmail();
+			const baseURI = 'https://analytics.google.com/analytics/web/';
 
-			if ( userEmail === undefined ) {
-				return undefined;
+			let appendedURL = baseURI;
+
+			if ( query ) {
+				appendedURL = addQueryArgs( baseURI, query );
 			}
 
-			const baseURI = 'https://analytics.google.com/analytics/web/';
-			const queryParams = query
-				? { ...query, authuser: userEmail }
-				: { authuser: userEmail };
-			const baseURIWithQuery = addQueryArgs( baseURI, queryParams );
 			if ( path ) {
 				const sanitizedPath = `/${ path.replace( /^\//, '' ) }`;
-				return `${ baseURIWithQuery }#${ sanitizedPath }`;
+				appendedURL = `${ appendedURL }#${ sanitizedPath }`;
 			}
-			return baseURIWithQuery;
+
+			const accountChooserBaseURI = select(
+				CORE_USER
+			).getAccountChooserURL( appendedURL );
+
+			if ( accountChooserBaseURI ) {
+				return accountChooserBaseURI;
+			}
+
+			return undefined;
```

Comment:
This isn't really a proper "return early" type of approach.

In this case, our "ideal" or "happy path" is not the final return value, which makes this a bit harder to parse. In most of our code we will return early when we don't have the proper amount of data to return the value a selector is designed to return. This makes it easier to follow the logic, but also allows us to comment out specific reasons why we "bail early" in a function and not add many conditions to the return of the "proper" value.

So this would be better written as:

```suggestion
			if ( accountChooserBaseURI === undefined ) {
				return undefined
			}

			return accountChooserBaseURI;
```